component-descriptor-v2-schema.yaml

```diff
@@ -82,6 +92,10 @@ definitions:
         type: 'string' # actually: component-type w/ special restrictions
       version:
         type: 'string' # actually: relaxed semver?
+      labels:
+        type: 'array'
```

Comment:
I thought that we wanted to have a map?
like
```yaml
labels:
  name1: value1
  name2: value2
```

Reply:
do we want all possible structures in the labels not just strings?

Reply:
it would be more flexible, if we allow objects..

Reply:
ok you're right allowing all types would be more flexible and otherwise people will encode data in strings.
There you also do not want a map but a list because of consistency with the other lists?

Reply:
yes, precisely. consistency on the one-hand-side, and (a bit) easier handling when passing around individual objects, because this way, a single element contains all attrs (well, name and value..). While the `dict`-approach looks more elegant, to my experience, this may cause headaches when processing, or passing-around.

Also, as previously discussed, a linter is required to detect conflicts (whereas with a list, we might even allow duplicate elements, but can detect them from a regular parsing result).

One disadvantage we can handle, namely the dereferentiation by name (`componentdescriptor://<name>/<resource-name>`). This would be very ugly when e.g. using a jq-style syntax. However, I do not see why we cannot simply allow the aforementioned dereferentiation syntax (in particular, since names still are guaranteed to be unique, so this will merely cost us a list traversal, which should not be considerably more expensive than a dict-key-access).

Reply:
makes sense. A dict is only more useful when we use some kind of templating as we can easily use a specific the entry without jq like search.
But in the landscaper we can have different formats so there is no issue from our side.

Then lets use a list of labels mit `name` and `label`

Do we allow duplicate label names?

Reply:
ok, then I am going to merge this PR. let us hope we can eventually finalise the schema :see_no_evil: